third_party/eigen3/gpu_packet_math.patch

```diff
@@ -40,3 +40,18 @@
      return res;
    }
  };
+diff --git a/Eigen/src/Core/arch/GPU/Half.h b/Eigen/src/Core/arch/GPU/Half.h
+index f87d8a1..a84c4c8 100644
+--- a/Eigen/src/Core/arch/GPU/Half.h
++++ b/Eigen/src/Core/arch/GPU/Half.h
+@@ -52,7 +52,9 @@ namespace half_impl {
+ #if !defined(EIGEN_HAS_GPU_FP16)
+ // Make our own __half_raw definition that is similar to CUDA's.
+ struct __half_raw {
+-  EIGEN_DEVICE_FUNC __half_raw() : x(0) {}
```

Comment:
we could use #if !__HIP__ to conditionally enable : x(0)

__HIP__ is 1 for hip-clang but 0 for hcc.

Reply:
@mshivama would you mind amend the PR to adopt this proposed change from @yxsamliu , so at least we can guarantee the incumbent HIP/HCC path stay as-is, while preserving desired behavior for hip-clang path.

Reply:
@yxsamliu , also I'd like to discuss the possibility using an LLVM opt pass to get around this issue.

@deven-amd and I just had an discussion, based on his experiences talking with Eigen maintainers it's likely the patch would ever be accepted upstream. Therefore it means we would always have IRs where global symbols on address space 3 come with initializers. What if we introduce a simple LLVM opt pass on AMDGPU backend to remove such initializers? The pass would be easy enough to implement, wouldn't cause source-level change, and only applies on AMDGPU target.

Reply:
based on offline discussion with @yxsamliu it seems changing Clang CodeGen may be a better route.

Reply:
@whchung  I am going to withdraw this PR